File: SofiForce.Models/Models/EntityModels/RepresentativeModel.cs
using System;

namespace Models
{
    public class RepresentativeModel
    {
        public int RepresentativeId { get; set; }
        public int? BranchId { get; set; }
        public int? UserId { get; set; }
        public int? SupervisorId { get; set; }
        public int? KindId { get; set; }
        public string RepresentativeCode { get; set; }
        public string RepresentativeNameAr { get; set; }
        public string RepresentativeNameEn { get; set; }
        public string Phone { get; set; }
        public string PhoneAlternative { get; set; }
        public bool? IsActive { get; set; }
        public int? DisplayOrder { get; set; }
        public bool? CanEdit { get; set; }
        public bool? CanDelete { get; set; }
        public DateTime? JoinDate { get; set; }
        public string Notes { get; set; }
        public string Icon { get; set; }
        public string Color { get; set; }
        public bool? IsTerminated { get; set; }
        public DateTime? TerminationDate { get; set; }
        public int? TerminationReasonId { get; set; }


        public string CompanyCode { get; set; }


    }
}

File: SofiForce.Web/ClientApp/src/app/core/Models/EntityModels/RepresentativeModel.ts
import { BranchModel } from "./branchModel";
import { AppUserModel } from "./appUserModel";
// import { SupervisorModel } from "./supervisorModel";
// import { RepresentativeKindModel } from "./representativeKindModel";
import { SalesOrderModel } from "./salesOrderModel";

export class RepresentativeModel {
    representativeId: number;
    branchId: number | null;
    branchCode: string | null;
    userId: number | null;
    supervisorId: number | null;
    kindId: number | null;
    companyCode: string;
    representativeCode: string;
    representativeNameAr: string;
    representativeNameEn: string;
    isActive: boolean | null;
    displayOrder: number | null;
    canEdit: boolean | null;
    canDelete: boolean | null;
    icon: string;
    color: string;
    cBy: number | null;
    cDate: Date | null;
    eBy: number | null;
    eDate: Date | null;

    isTerminated: boolean | null;
    phone : string | null;
    phoneAlternative : string | null;
    terminationDate : Date | null;
    terminationReasonId : number | null;
    businessUnitId : number | null;

    joinDate:Date |null;
    supervisorCode: string;
    
    notes:string | null
}